### PR TITLE
improvement: introduce tenant_id in JWT CustomClaims

### PR DESCRIPTION
**Pull request by Junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Introduce tenant_id in JWT CustomClaims

## Changelog

- Improvement

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/4936

### CODE REVIEW

**Review by youngsofun — APPROVED** [review]

File: query/src/servers/http/v1/http_query_handlers.rs [hunk 1]
```diff
@@ -142,12 +138,11 @@ pub(crate) struct CancelParams {
 
 #[poem::handler]
 async fn query_cancel_handler(
-    sessions_extension: Data<&Arc<SessionManager>>,
+    ctx: &HttpQueryContext,
```
Comment (flaneur2020) [inline query/src/servers/http/v1/http_query_handlers.rs:141]: 6666, nice abstraction!

File: query/src/sessions/session_ctx.rs [hunk 2]
```diff
@@ -77,10 +79,21 @@ impl SessionContext {
         *lock = db
     }
 
-    pub fn get_tenant(&self) -> String {
+    pub fn get_current_tenant(&self) -> String {
+        if self.conf.query.management_mode {
```
Comment (flaneur2020) [inline query/src/sessions/session_ctx.rs:83]: add a comment about the difference between the current tenant logic of management_mode and normal case?

File: query/src/sessions/session_settings.rs [hunk 3]
```diff
@@ -300,7 +300,7 @@ impl Settings {
     }
 
     fn set_to_global(&self, setting: &mut SettingValue) -> Result<()> {
-        let tenant = self.session_ctx.get_tenant();
```
Comment (flaneur2020) [inline query/src/sessions/session_settings.rs:303]: how about remove the `get_tenant()` method of session_ctx struct? 

newcomers may miss the different uses between the two methods 🤔

File: query/src/users/auth/jwt/authenticator.rs [hunk 4]
```diff
@@ -35,18 +35,26 @@ pub struct JwtAuthenticator {
...
 impl CustomClaims {
     pub fn new() -> Self {
-        CustomClaims { ensure_user: None }
+        CustomClaims {
+            tenant_id: None,
+            ensure_user: None,
+        }
+    }
+
+    pub fn with_tenant_id(mut self, tenant_id: &str) -> Self {
```
Comment (flaneur2020) [inline query/src/users/auth/jwt/authenticator.rs:55]: how about move a `tenant_id: String` here? maybe this could reduce a string copy

Comment (Junnplus) [reply query/src/users/auth/jwt/authenticator.rs:55]: This is for testing only.

**Pull request merged**